cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/FileSystemUtil.java

```diff
@@ -102,4 +107,47 @@ public static boolean isZipFile(URI uri) {
       throw new CobiGenRuntimeException("Unable to read file " + file.getAbsolutePath(), e);
     }
   }
+
+  /**
+   * collect all context.xml from a given path
+   *
+   * @param templatesPath the path to a templates location or any given path to search for
+   * @return a path list of all found context.xml
+   */
+  public static List<Path> collectAllContextXML(Path templatesPath) {
+
+    if (templatesPath == null)
+      throw new CobiGenRuntimeException(
+          "The templatespath cannot be null. Please try again with a proper templatespath!");
+    List<Path> result = null;
+    try (Stream<Path> pathStream = Files.find(templatesPath, Integer.MAX_VALUE, (p, basicFileAttributes) -> p
+        .getFileName().toString().equalsIgnoreCase(ConfigurationConstants.CONTEXT_CONFIG_FILENAME))) {
+      result = pathStream.collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new CobiGenRuntimeException("An error occured while collecing the context.xml files!");
+    }
+    return result;
```

Comment:
This is basically a duplication of collectAllTemplatesXML. Please refactor.

Reply:
done